services/worker-manager/schemas/v1/config-aws.yml

```diff
@@ -25,14 +25,27 @@ properties:
             ImageId:
               type: string
               description: The ID of the AMI to launch an instance from.
+            # These properties are not allowed, so we assert that
+            # they do not match the empty schema
+            TagSpecifications: {not: {}}
+            IamInstanceProfile: {not: {}}
```

Comment:
Can you talk more about what you saw, and how you came up with this solution? (`{not: {}}` instead of `not.required:`)

Reply:
If you have the schema:
```yaml
not:
    required: [a, b]
```
then `{a: stuff}` and `{b: stuff}` will both validate, but `{'a: stuff, b:stuff}` will not. That is because both the first two do *not* validate against `required: [a, b]` (since in either case they are missing one of the required properties).

The way that `{not: {}}` works is that `{}` will match against any value (including `null`). So, `not: {}` will not match against any value. Therefore, if an optional property has to match `not: {}`, it can't be specified (since the value specified would not match the schema).